heft-plugins/heft-jest-plugin/src/transformers/StringMockTransformer.ts

```diff
@@ -0,0 +1,22 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import type { SyncTransformer, TransformedSource, TransformOptions } from '@jest/transform';
+import type { Config } from '@jest/types';
+
+/**
+ * This Jest transform handles imports of data files (e.g. .png, .jpg) that would normally be
+ * processed by a Webpack's file-loader. Instead of actually loading the resource, we return the file's name.
+ * Webpack's file-loader normally returns the resource's URL, and the filename is an equivalent for a Node
+ * environment.
+ */
+export class StringMockTransformer implements SyncTransformer {
+  public process(sourceText: string, sourcePath: Config.Path, options: TransformOptions): TransformedSource {
+    // Double-escape "'" and "\" characters in the filename because this is going to be serialized
+    // in a string in generated JS code, bounded by single quotes
+    const escapedSourcePath: string = sourcePath.replace(/\'/g, "\\'").replace(/\\/g, '\\\\');
+
+    // For a file called "myImage.png", this will generate a JS module that exports the literal string "myImage.png"
+    return `module.exports = '${escapedSourcePath}';`;
```

Comment:
```suggestion
    // For a file called "myImage.png", this will generate a JS module that exports the literal string "myImage.png"
    return `module.exports = ${JSON.stringify(sourcePath)};`;
```

Any reason not to just do that?

Reply:
I don't see why not. It should be just as valid.